normandy/recipes/models.py

```diff
@@ -122,18 +163,52 @@ def update_signature(self):
 
     @transaction.atomic
     def update(self, force=False, **data):
```

Comment:
Also, for finding matching recipe revisions, it seems to make more sense to include the relations in the revision ID hash, and just create a new revision and look for revisions with that hash.